src/waypoint.jsx

```diff
@@ -295,8 +311,9 @@ export default class Waypoint extends React.Component {
 
   _getBounds() {
     const horizontal = this.props.horizontal;
-    const waypointTop = horizontal ? this._ref.getBoundingClientRect().left :
-      this._ref.getBoundingClientRect().top;
+    const { left, top, right, bottom } = this._ref.getBoundingClientRect();
+    const waypointTop = horizontal ? left : top;
+    const waypointBottom = horizontal ? right : bottom;
```

Comment:
top/bottom is very vertically oriented terminology. I know you were just following the local pattern here, but what do you think about using `waypointStart` and `waypointEnd` instead?